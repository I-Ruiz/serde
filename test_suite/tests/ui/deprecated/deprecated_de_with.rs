#![deny(deprecated)]

use serde::Deserializer;
use serde_derive::Deserialize;

#[derive(Deserialize)]
pub struct Struct {
    #[serde(deserialize_with = "deprecated_with")]
    pub field: i32,
}

#[deprecated]
fn deprecated_with<'de, D>(_deserializer: D) -> Result<i32, D::Error>
where
    D: Deserializer<'de>,
{
    unimplemented!()
}

fn main() {}
