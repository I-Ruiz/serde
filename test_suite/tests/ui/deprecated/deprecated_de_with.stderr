error: use of deprecated function `deprecated_with`
 --> tests/ui/deprecated/deprecated_de_with.rs:8:32
  |
8 |     #[serde(deserialize_with = "deprecated_with")]
  |                                ^^^^^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> tests/ui/deprecated/deprecated_de_with.rs:1:9
  |
1 | #![deny(deprecated)]
  |         ^^^^^^^^^^
